EpsilonLights/Tests/tests/LightsTest/LightsTest.c

```diff
@@ -0,0 +1,391 @@
+#include "unity.h"
+
+#include "Mockstm32f4xx_hal_gpio.h"
+#include "Mockstm32f4xx_hal_can.h"
+//#include "Mockstm32f4xx_hal_adc.h"
+#include "Mockcmsis_os.h"
+
+//#include "stm32f4xx_hal_can.h"
+#include "Lights.h"
+#include "LightsTest.h"
+#include <stdio.h>
+
+lightsInfo lightCharacteristics = {0};
+
+uint8_t lightsInputs;
+uint8_t auxBmsInputs[2];
+uint8_t batteryErrors[5];
+uint8_t driversInputs[4];
+SigLightsHandle sigLightsHandle;
+
+osMutexId* canHandleMutex;
+
+void runLightsTests()
+{
+    RUN_TEST(test_updateLights1_headLightsOffBrakesLightsOffHazardsOffBmsStrobeLightsOff);
+    RUN_TEST(test_updateLights1_headLightsLowBreakLightsOnHazardsOnBmsStrobeLightOn);
+    RUN_TEST(test_updateLights1_headLightsHighHeadLightsLowBreakLightsOnHazardsOnBmsStrobeLightOn);
+    RUN_TEST(test_blinkSignalLights_sigLightsLeftOFFSigLightsRightOFF);
+    RUN_TEST(test_blinkSignalLights_sigLightsLeftONSigLightsRightONPrevSigStateZero);
+    RUN_TEST(test_blinkSignalLights_prevSigState1SigLightsLeftONSigLightsRightON);
+    RUN_TEST(test_updateStrobeLights1_ESTROBEONUpdateBlinkerTimer);
+    RUN_TEST(test_updateStrobeLights1_ESTROBEOFFResetBlinkerTimer);
+    RUN_TEST(test_updateStrobeLights1_ESTROBEOFFUpdateBlinkerTimer);
+    RUN_TEST(test_reportLightsToCan);
+    RUN_TEST(test_sendHeartbeat);
+    RUN_TEST(test_HAL_CAN_RxCpltCall_backLightsInputStdId);
+    RUN_TEST(test_HAL_CAN_RxCpltCallback_batteryStatErrorsStdId);
+    RUN_TEST(test_HAL_CAN_RxCpltCallback_driversInputsStdId);
+    RUN_TEST(test_HAL_CAN_RxCpltCallback_auxBmsInputStdId);
+    RUN_TEST(test_HAL_CAN_RxCpltCallback_batteryStatErrorsStdIdANDToggleRedPin);
+}
+
+void test_updateLights1_headLightsOffBrakesLightsOffHazardsOffBmsStrobeLightsOff()
+{
+    lightsInputs = 0b00000001;
+    driversInputs[1] = 0;
+    driversInputs[2] = 0;
+    auxBmsInputs[1] = 0;
+    driversInputs[3] = 0b00000000;
+    uint32_t prevWakeTime = 0;
+    lightCharacteristics.leftSignal = 1;
+    lightCharacteristics.rightSignal = 0;
```

Comment:
Style thing but the `uint32_t prevWakeTime` here looks odd with the other ones. Perhaps put it at the top and separate it from this block? (Remember to update this for all the test cases that has this)